Intersect.Server/Entities/Player.cs

```diff
@@ -3646,52 +3648,90 @@ public void CraftItem(Guid id)
                     }
                 }
 
-                //Take the items
-                foreach (var c in CraftBase.Get(id).Ingredients)
+                //Return true if the craft was a success
+                if (Randomization.Next(1, 101) > mCraftItem.FailureChance)
                 {
-                    if (!TryTakeItem(c.ItemId, c.Quantity))
+                    //Take the items
+                    foreach (var c in mCraftItem.Ingredients)
+                    {
+                        if (!TryTakeItem(c.ItemId, c.Quantity))
+                        {
+                            for (var i = 0; i < invbackup.Count; i++)
+                            {
+                                Items[i].Set(invbackup[i]);
+                            }
+
+                            PacketSender.SendInventory(this);
+                            CraftId = Guid.Empty;
+
+                            return;
+                        }
+                    }
+
+                    //Give them the craft
+                    var quantity = Math.Max(mCraftItem.Quantity, 1);
+                    var itm = ItemBase.Get(mCraftItem.ItemId);
```

Comment:
instead of "itm", try with "craftItemId" for this local variable's name.

Reply:
Uh, it shouldn't be craftItemId, it's not an id, it's the descriptor. `item` is better than `itm` though.